copy/copy.go

```diff
@@ -1353,7 +1374,11 @@ func (c *copier) copyBlobFromStream(ctx context.Context, srcStream io.Reader, sr
 		compressionOperation = types.PreserveOriginal
 		inputInfo = srcInfo
 		uploadCompressorName = srcCompressorName
-		uploadCompressionFormat = nil
+		if isCompressed {
```

Comment:
Please document that this, as a side effect, causes a MIME type update if `LayerInfosForCopy` replaces a blob, because `LayerInfosForCopy`’s `MediaType` update does not cause a manifest edit.

Reply:
Added `Remember if the original blob was compressed, and if so how, so that if LayerInfosForCopy() returned something that differs from what was in the source's manifest, and UpdatedImage() needs to call UpdateLayerInfos(), it will be able to correctly derive the MediaType for the copied blob.`